apps/web/src/components/CookieBanner/CookieBanner.tsx

```diff
@@ -0,0 +1,163 @@
+import { getCookie, setCookie } from 'cookies-next'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import type { MouseEventHandler } from 'react'
+import { useEffect, useRef, useState } from 'react'
+
+import {
+  COOKIE_SETTINGS_CHANGE_EVENT,
+  SE_GDPR_COOKIE_ACCEPT_VALUE,
+  SE_GDPR_COOKIE_NAME,
+  SE_GDPR_COOKIE_REJECT_VALUE,
+} from '@/constants/cookies'
+import { getGDPRCookieExpiryDate } from '@/utils/date'
+
+enum CookieBannerView {
+  Hidden,
+  Selection,
+  Accepted,
+  Rejected,
+}
+
+export function CookieBanner() {
+  const [view, setView] = useState(CookieBannerView.Hidden)
+
+  const regionRef = useRef<HTMLDivElement>(null)
+
+  const router = useRouter()
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      if (url.includes('?change-settings=1')) {
+        setView(CookieBannerView.Selection)
+        regionRef.current?.focus()
+      } else {
+        // Hide banner after navigating away from cookie policy page
+        const isCookieSet = Boolean(getCookie(SE_GDPR_COOKIE_NAME))
+        if (view === CookieBannerView.Selection && isCookieSet) {
+          setView(CookieBannerView.Hidden)
+        }
+      }
+    }
+    router.events.on('routeChangeComplete', handleRouteChange)
+    return () => {
+      router.events.off('routeChangeComplete', handleRouteChange)
+    }
+  }, [router, view])
+
+  useEffect(() => {
```

Comment:
I'm concerned about the number of potential re-renders for the `useEffects`, your `view` variable is in your dependancy array which you are then updating in your `useEffect` which will fire a re-render and so on, also 2 `useEffect` with `view` in their dependancy array, who "wins"